src/cpufreq.c

```diff
@@ -132,34 +132,36 @@ static int cpufreq_read(void) {
     /* convert kHz to Hz */
     v.gauge *= 1000.0;
 
-    cpufreq_submit(i, "cpufreq", NULL, &v);
+    cpufreq_submit(cpu, "cpufreq", NULL, &v);
 
     if (report_p_stats) {
       /* Read total transitions for cpu frequency */
       snprintf(filename, sizeof(filename),
-               "/sys/devices/system/cpu/cpu%d/cpufreq/stats/total_trans", i);
+               "/sys/devices/system/cpu/cpu%d/cpufreq/stats/total_trans", cpu);
       if (parse_value_file(filename, &v, DS_TYPE_COUNTER) != 0) {
```

Comment:
Please use `DS_TYPE_DERIVE` instead of counter.

Reply:
The same is here, as described above.

Number of frequency transitions is a _counter_.

File `linux-3.2.78/drivers/cpufreq/cpufreq_stats.c` has following structure:
```
struct cpufreq_stats {
    unsigned int cpu;
    unsigned int total_trans;
    unsigned long long  last_time;
    unsigned int max_state;
    unsigned int state_num;
    unsigned int last_index;
    cputime64_t *time_in_state;
    unsigned int *freq_table;
#ifdef CONFIG_CPU_FREQ_STAT_DETAILS
    <------>unsigned int *trans_table;
#endif
};
```

When use DS_TYPE_DERIVE, `parse_value()` handles value as signed.

Reply:
That's fine, 2.9 billion years is well outside of my life-span so I'm not going to care ;)